src/core/server/saved_objects/export/collect_exported_objects.ts

```diff
@@ -126,3 +165,68 @@ const fetchReferences = async ({
       .map((obj) => ({ type: obj.type, id: obj.id })),
   };
 };
+
+const buildTransforms = (typeRegistry: ISavedObjectTypeRegistry) =>
+  typeRegistry.getAllTypes().reduce((transforms, type) => {
+    if (type.management?.onExport) {
+      return {
+        ...transforms,
+        [type.name]: type.management.onExport,
+      };
+    }
+    return transforms;
+  }, {} as Record<string, SavedObjectsExportTransform>);
+
+const buildIsExportable = (
+  typeRegistry: ISavedObjectTypeRegistry
+): SavedObjectsExportablePredicate<any> => {
+  const exportablePerType = typeRegistry.getAllTypes().reduce((transforms, type) => {
+    if (type.management?.isExportable) {
+      transforms[type.name] = type.management.isExportable;
+    }
+    return transforms;
+  }, {} as Record<string, SavedObjectsExportablePredicate>);
+
+  return (obj: SavedObject) => {
+    const typePredicate = exportablePerType[obj.type];
+    return typePredicate ? typePredicate(obj) : true;
+  };
+};
+
+const splitByExportability = async (
+  objects: SavedObject[],
+  isExportable: SavedObjectsExportablePredicate<any>
+) => {
+  const exportableObjects: SavedObject[] = [];
+  const nonExportableObjects: SavedObject[] = [];
+  await Promise.all(
+    objects.map(async (obj) => {
```

Comment:
A large number of concurrent promises can place undue burden on the Kibana process. We've used `p-map` in other places to batch these:

https://github.com/elastic/kibana/blob/238791b9421cda571011fd416ae0ffd090b1e94d/src/core/server/saved_objects/import/lib/check_origin_conflicts.ts#L135-L140

Reply:
See https://github.com/elastic/kibana/pull/101860#discussion_r651756274